### PR TITLE
enlightenment: acpid window is not shown on 32-bit arm

### PR DESCRIPTION
**Pull request by ggardet**
- Verification run: https://openqa.opensuse.org/t1787159

### CODE REVIEW

File: tests/x11/enlightenment_first_start.pm [hunk 1]
```diff
@@ -43,7 +44,7 @@ sub run {
         assert_screen 'enlightenment_compositing';
     }
     assert_and_click "enlightenment_assistant_next";
-    assert_and_click 'enlightenment_acpid_missing';
+    assert_and_click 'enlightenment_acpid_missing' if !is_arm;
```
Comment (Vogtinator) [inline tests/x11/enlightenment_first_start.pm:47]:
```suggestion
    assert_and_click 'enlightenment_acpid_missing' unless is_arm;
```

**Comment by Vogtinator** [comment]
It's a bit weird that it shows up everywhere except for 32bit arm. Maybe `acpid` is installed?

File: tests/x11/enlightenment_first_start.pm (hunk 1 above)
Comment (ggardet) [reply tests/x11/enlightenment_first_start.pm]: Done

**Comment by ggardet** [comment]
> It's a bit weird that it shows up everywhere except for 32bit arm. Maybe `acpid` is installed?

Or there is no ACPI in qemu for 32-bit arm?

**Comment by Vogtinator** [comment]
> > It's a bit weird that it shows up everywhere except for 32bit arm. Maybe `acpid` is installed?
> 
> Or there is no ACPI in qemu for 32-bit arm?

Maybe. Enlightenment checks for `/proc/acpi/`: https://github.com/Enlightenment/enlightenment/blob/92695f10fcdf042fff9d2e3e9d8f60a96fa16ce4/src/bin/e_acpi.c#L155

**Comment by Vogtinator** [comment]
> > > It's a bit weird that it shows up everywhere except for 32bit arm. Maybe `acpid` is installed?
> > 
> > 
> > Or there is no ACPI in qemu for 32-bit arm?
> 
> Maybe. Enlightenment checks for `/proc/acpi/`: https://github.com/Enlightenment/enlightenment/blob/92695f10fcdf042fff9d2e3e9d8f60a96fa16ce4/src/bin/e_acpi.c#L155

I couldn't find anything which disables ACPI for 32bit ARM in either QEMU or OVMF. What's the output of `ls -R /sys/firmware/qemu_fw_cfg/by_name`? If QEMU exposes ACPI, then the tables should be in there.

**Comment by ggardet** [comment]
> > > > It's a bit weird that it shows up everywhere except for 32bit arm. Maybe `acpid` is installed?
> > > 
> > > 
> > > Or there is no ACPI in qemu for 32-bit arm?
> > 
> > 
> > Maybe. Enlightenment checks for `/proc/acpi/`: https://github.com/Enlightenment/enlightenment/blob/92695f10fcdf042fff9d2e3e9d8f60a96fa16ce4/src/bin/e_acpi.c#L155
> 
> I couldn't find anything which disables ACPI for 32bit ARM in either QEMU or OVMF. What's the output of `ls -R /sys/firmware/qemu_fw_cfg/by_name`? If QEMU exposes ACPI, then the tables should be in there.

There is no `/proc/acpi` and `ls /sys/firmware/` returns `devicetree  dmi  efi  fdt`

**Comment by Vogtinator** [comment]
Hm, that means qemu_fw_cfg doesn't work on 32bit arm at all?

**Comment by ggardet** [comment]
> Hm, that means qemu_fw_cfg doesn't work on 32bit arm at all?

It seems there is no ACPI option available in armv7 kernel config.

**Comment by Vogtinator** [comment]
Indeed, `ARCH_SUPPORTS_ACPI` is only set for x86, ia64 and arm64.

**Comment by Vogtinator** [comment]
> Indeed, `ARCH_SUPPORTS_ACPI` is only set for x86, ia64 and arm64.

Which means that maybe `if (get_required_var('ARCH') =~ /86/ || is_aarch64)` is more appropriate

**Comment by ggardet** [comment]
> > Indeed, `ARCH_SUPPORTS_ACPI` is only set for x86, ia64 and arm64.
> 
> Which means that maybe `if (get_required_var('ARCH') =~ /86/ || is_aarch64)` is more appropriate

Yes, updated accordingly.

**Review by Vogtinator — APPROVED** [review]
Restarted VR, is green: https://openqa.opensuse.org/tests/1788716#details

**Pull request merged**